ibis/backends/postgres/__init__.py

```diff
@@ -184,53 +194,102 @@ def list_databases(self, like=None):
+  string_to_array(pg_catalog.pg_get_function_arguments(p.oid), ', ') as signature,
+  CASE p.prokind
+    WHEN 'a' THEN 'agg'
+    WHEN 'w' THEN 'window'
+    WHEN 'p' THEN 'proc'
+    ELSE 'func'
+  END as "Type"
+FROM pg_catalog.pg_proc p
+LEFT JOIN pg_catalog.pg_namespace n ON n.oid = p.pronamespace
+WHERE p.proname = '{name}'
+""" + f"AND n.nspname OPERATOR(pg_catalog.~) '^({schema})$' COLLATE pg_catalog.default" * (
+            schema is not None
+        )
 
-        Parameters
-        ----------
-        pyfunc
-            Python function
-        in_types
-            Input types
-        out_type
-            Output type
-        schema
-            The postgres schema in which to define the UDF
-        replace
-            replace UDF in database if already exists
-        name
-            name for the UDF to be defined in database
-        language
-            Language extension to use for PL/Python
-
-        Returns
-        -------
-        Callable
-            A callable ibis expression
-
-        Function that takes in Column arguments and returns an instance
-        inheriting from PostgresUDFNode
-        """
+        def split_name_type(arg: str) -> tuple[str, dt.DataType]:
+            name, typ = arg.split(" ", 1)
+            return name, _get_type(typ)
 
-        return _udf(
-            client=self,
-            python_func=pyfunc,
-            in_types=in_types,
-            out_type=out_type,
-            schema=schema,
-            replace=replace,
+        with self.begin() as con:
+            rows = con.exec_driver_sql(query).mappings().fetchall()
+
+            if not rows:
+                name = f"{schema}.{name}" if schema else name
+                raise exc.MissingUDFError(name)
+            elif len(rows) > 1:
+                raise exc.AmbiguousUDFError(name)
+
+            [row] = rows
+            return_type = _get_type(row["return_type"])
+            signature = list(map(split_name_type, row["signature"]))
+
+        # dummy callable
+        def fake_func(*args, **kwargs):
+            ...
+
+        fake_func.__name__ = name
+        fake_func.__signature__ = inspect.Signature(
+            [
+                inspect.Parameter(
+                    name, kind=inspect.Parameter.POSITIONAL_OR_KEYWORD, annotation=typ
+                )
+                for name, typ in signature
+            ],
+            return_annotation=return_type,
+        )
+        fake_func.__annotations__ = {"return": return_type, **dict(signature)}
+        op = ops.udf.scalar(fake_func, schema=schema, __is_server_defined__=True)
+        return op
+
+    def _get_udf_source(self, udf_node: ops.SingleScalarUDF):
+        config = udf_node.__config__["kwargs"]
+        immutable = config.get("immutable", False) * "IMMUTABLE"
+        strict = config.get("strict", False) * "STRICT"
+        leakproof = config.get("leakproof", False) * "LEAKPROOF"
+        parallel = config.get("parallel", "UNSAFE").upper()
```

Comment:
I may kill all of these options, they're kind of noisy and I am not sure what their effect on performance is. In theory enabling them improves performance but adding these at the start might be overkill.

Reply:
Keeping things simpler for now makes sense. Can always add them back later if/when someone asks for it.

Reply:
Removed.